$OpenBSD: patch-gnome-session_main_c,v 1.10 2012/09/28 07:34:24 ajacoutot Exp $

Set XDG_MENU_PREFIX so that menu is created from
SYSCONFDIR/xdg/menus/gnome-applications.menu

--- gnome-session/main.c.orig	Mon Sep 24 20:07:50 2012
+++ gnome-session/main.c	Fri Sep 28 09:24:16 2012
@@ -347,6 +347,13 @@ main (int argc, char **argv)
          */
         gsm_util_setenv ("GNOME_DESKTOP_SESSION_ID", "this-is-deprecated");
 
+        /* Set default menu to gnome-applications.menu if XDG_MENU_PREFIX
+         * is not set.
+         */
+        if (!g_getenv ("XDG_MENU_PREFIX")) {
+                gsm_util_setenv ("XDG_MENU_PREFIX", "gnome-");
+        }
+
         client_store = gsm_store_new ();
 
         xsmp_server = gsm_xsmp_server_new (client_store);
